redis/pool_test.go

```diff
@@ -368,21 +396,92 @@ func TestPoolConcurrenSendReceive(t *testing.T) {
 }
 
 func TestPoolBorrowCheck(t *testing.T) {
-	d := poolDialer{t: t}
-	p := &redis.Pool{
-		MaxIdle:      2,
-		Dial:         d.dial,
-		TestOnBorrow: func(redis.Conn, time.Time) error { return redis.Error("BLAH") },
+	pingN := func(ctx context.Context, p *redis.Pool, n int) {
+		for i := 0; i < n; i++ {
+			c, err := p.GetContext(ctx)
+			require.NoError(t, err)
+			_, err = redis.DoContext(c, ctx, "PING")
+			require.NoError(t, err)
+			require.NoError(t, c.Close())
+		}
 	}
-	defer p.Close()
 
-	for i := 0; i < 10; i++ {
-		c := p.Get()
-		_, err := c.Do("PING")
-		require.NoError(t, err)
-		c.Close()
-	}
-	d.check("1", p, 10, 1, 0)
+	t.Run("TestOnBorrow returns error", func(t *testing.T) {
```

Comment:
Can we keep the test names short without spaces yet still descriptive e.g. `TestOnBorrow-error` this makes the names searchable and the output concise.